tripal/src/Form/TripalImporterForm.php

```diff
@@ -256,16 +256,43 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
         $file_existing = $form_values['file_upload_existing'];
       }
     }
+
+    // Remote file validation.
     if (array_key_exists('file_remote', $importer_def) and $importer_def['file_remote'] == TRUE) {
       $file_remote = trim($form_values['file_remote']);
+      if ($file_remote) {
+        // Validate that the remote URI is of the correct format.
+        if (filter_var($file_remote, FILTER_VALIDATE_URL) === false) {
+          $form_state->setErrorByName('file_remote', t('The Remote Path provided is not a valid URI.'));
+        }
+
+        // Validate a correct format remote URI to make sure it can be accessed.
+        else {
+          $headers = @get_headers($file_remote);
+          if (($headers === false) or (!is_array($headers)) or (!strpos($headers[0], '200'))) {
+            $form_state->setErrorByName('file_remote', t('The Remote Path provided cannot be accessed.'
+              . ' Check that it is correct.'));
+          }
+          // Some internet providers will insert content with a valid '200' response code for invalid
+          // URIs that redirects to their own search page, such as AT&T "DNS Error Assist".
+          // Try to detect this by rejecting 'http-equiv="refresh"' in the first 1000 bytes
```

Comment:
I cannot think of a good, repeatable way to test this 🤔 and would be tempted to remove it... Alternatively, we could test the the downloaded header indicates the right file format for the importer but this feels a bit finicky as well.

Reply:
I also feel very uncertain about this bit of code, as you note, it is nearly impossible to test, and may not handle any situation other than my specific internet provider. 
And yes, each importer could check the header when the job is run, that is probably much safer and reliable.

I shall remove it 💣